src/qtt/algorithms/functions.py

```diff
@@ -471,3 +471,60 @@ def logistic(x, x0=0, alpha=1):
     """
     f = 1 / (1 + np.exp(-2 * alpha * (x - x0)))
     return f
+
+
+def raised_cosine_frequency_domain(f: np.ndarray, beta: float, T: float) -> np.ndarray:
+    """ Raised cosine frequency domain function
+
+    See https://en.wikipedia.org/wiki/Raised-cosine_filter
+
+    Args:
+        t: Frequency
+        beta: Roll-off factor
+        T: Symbol period
+
+    Returns:
+        Calculated values of the raised cosine
+    """
+    two_T = 2*T
+    if beta == 0:
+        return np.abs(f) < 1/(two_T)
+    ww = np.abs(f)-(1-beta)/(two_T)
+    value = .5*(1+np.cos((np.pi*T/beta)*ww))
+    idx = np.abs(f) <= (1-beta)/two_T
+    value[idx] = 1
```

Comment:
same as a later comment, idx is boolean value. I guess you use implicit casting but even then for me it is strange to read.

idx = 1 if np.abs(f) <= (1-beta)/two_T else 0

something like this?

Reply:
Same as above. Note that scalar input values are not allowed here, so `raised_cosine(1.1, .2, .4)` will fail. (for this reason the argument is annotated as `np.ndarray`, and not ` npt.ArrayLike`)